nixos/modules/services/networking/dnsproxy.nix

```diff
@@ -67,19 +79,42 @@ in
       '';
     };
 
+    # Allows files to be securely passed in command-line flags through systemd credentials
+    secretFlags = mkOption {
```

Comment:
I don't love the name of this option, but can't think of anything better. Suggestions are welcome